contrib/autopilot/autopilot.py

```diff
@@ -0,0 +1,902 @@
+                    source_out.write(line)
+            with third_party.bin_open(target_fname) as inp:
+                for line in inp:
+                    line = (re.sub(r"\s", " ", line.decode("utf-8"))).encode("utf-8") + b"\n"
+                    target_out.write(line)
+
+
+def plain_text_iter(fname: str, text_type: str, data_side: str) -> Iterable[str]:
+    """
+    Extract plain text from file as iterable.  Also take steps to ensure that
+    whitespace characters (including unicode newlines) are normalized and
+    outputs are line-parallel with inputs considering ASCII newlines only.
+
+    :param fname: Path of possibly gzipped input file.
+    :param text_type: One of TEXT_*, indicating data format.
+    :param data_side: DATA_SRC or DATA_TRG.
+    """
+    if text_type in (TEXT_UTF8_RAW, TEXT_UTF8_TOKENIZED):
+        with third_party.bin_open(fname) as inp:
+            for line in inp:
+                line = re.sub(r"\s", " ", line.decode("utf-8"))
+                yield line.strip()
+    elif text_type == TEXT_UTF8_RAW_SGML:
+        with third_party.bin_open(fname) as inp:
+            for line in inp:
+                line = re.sub(r"\s", " ", line.decode("utf-8"))
+                if line.startswith("<seg "):
+                    # Extract segment text
+                    text = re.sub(r"<seg.*?>(.*)</seg>.*?", "\\1", line)
+                    text = re.sub(r"\s+", " ", text.strip())
+                    # Unescape XML entities
+                    text = text.replace("&quot;", "\"")
+                    text = text.replace("&apos;", "'")
+                    text = text.replace("&lt;", "<")
+                    text = text.replace("&gt;", ">")
+                    text = text.replace("&amp;", "&")
+                    yield text
+    elif text_type in (TEXT_UTF8_RAW_BITEXT, TEXT_UTF8_RAW_BITEXT_REVERSE):
+        # Select source or target field, reversing if needed
+        if text_type == TEXT_UTF8_RAW_BITEXT:
+            field_id = 0 if data_side == DATA_SRC else 1
+        else:
+            field_id = 1 if data_side == DATA_SRC else 0
+        with third_party.bin_open(fname) as inp:
+            for line in inp:
+                line = re.sub(r"\s", " ", line.decode("utf-8"))
+                fields = line.split("|||")
+                yield fields[field_id].strip()
+    else:
+        raise RuntimeError("Unknown text type: %s" % text_type)
+
+
+def touch_file(fname: str):
+    """Create a file if not present, update access time."""
+    # Reference not needed since there will be no reads or writes
+    with open(fname, "a"):
+        os.utime(fname, None)
+
+
+def renew_step_dir(step_dir: str):
+    """Delete step directory if exists and create, reporting actions."""
+    if os.path.exists(step_dir):
+        logging.info("Remove unfinished step %s", step_dir)
+        shutil.rmtree(step_dir)
+    logging.info("Create: %s", step_dir)
+    os.makedirs(step_dir)
+
+
+def call_sockeye_train(args: List[str],
```

Comment:
Not relevant for this PR, but this task may benefit from the recently added feature of using config files and being able to start training as an API call (`sockeye.train.train()` with parsed args directly).

Reply:
That's a very good idea.  We can follow up in another PR.